maskinporten_api/jwt_client.py

```diff
@@ -0,0 +1,133 @@
+"""Module for obtaining access tokens for Maskinporten.
+
+The main entry point is the `JWTAuthClient.get_access_token` method. Example
+usage:
+
+    conf = JWTConfig(
+        issuer=my_issuer,
+        consumer_org=my_consumer_org,
+        certificate=my_certificate,
+        private_key=my_private_key,
+    )
+    client = JWTAuthClient(conf, my_well_known_endpoint)
+    access_token = client.get_access_token(["list", "of", "scopes"])
+"""
+
+import base64
+import json
+import logging
+import uuid
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+import jwt
+import requests
+from OpenSSL import crypto
+
+
+class JWTAuthError(Exception):
+    pass
+
+
+@dataclass
+class JWTConfig:
+    issuer: str
+    consumer_org: str
+    certificate: crypto.X509
+    private_key: crypto.PKey
+
+
+@dataclass
```

Comment:
Burde den her være en `@dataclass`? Vet ikke hva som er fasit her, men jeg har inntrykk av at dataklasser er mest ment for klasser som holder på verdier mer enn å utføre operasjoner. Ikke at det har noen teknisk betydning her, men når jeg ser en `@dataclass` så forventer jeg ikke at den inneholder mye funksjonalitet utover å representere en datastruktur

Reply:
Ja, tror jeg er enig med deg, denne klassen ble litt feit. Gjør den om til en vanlig klasse.